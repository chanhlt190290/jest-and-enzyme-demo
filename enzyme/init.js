import raf from './temp-polyfill';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
