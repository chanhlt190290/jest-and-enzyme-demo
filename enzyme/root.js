var  jsdom = require('jsdom');
const {JSDOM}  = jsdom;

const {document}  = (new JSDOM('<html><body></body></html>')).window; 
require('babel-register')();

var exposedProperties = ['window', 'navigator', 'document'];

global.document = document;
global.window = document.defaultView;
Object.keys(document.defaultView).forEach((property) => {
  if (typeof global[property] === 'undefined') {
    exposedProperties.push(property);
    global[property] = document.defaultView[property];
  }
});

global.navigator = {
  userAgent: 'node.js'
};

documentRef = document;

