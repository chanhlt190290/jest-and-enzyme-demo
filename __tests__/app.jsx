import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/app.jsx';
import expect from 'expect';
import {mount, shallow, configure} from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });


describe('App', () => {
  let app;

  beforeEach(() => {
    app = shallow(<App />);
  });

  it('App snapshot', () => {
    const component = renderer.create(<App />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('App renders nested components', () => {
    expect(app.find('Add').length).toEqual(1);
    expect(app.find('List').length).toEqual(1);
  });

  it('onAdd updates List', () => {
    const add = app.find('Add').first();
    add.props().onAdd('Name 1');
    app.update();     
    const list = app.find('List').first();
    const listData = list.props().data;
    expect(listData.length).toEqual(1);
    expect(listData[0]).toEqual('Name 1');
  });
});