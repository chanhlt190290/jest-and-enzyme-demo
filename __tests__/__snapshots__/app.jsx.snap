// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App snapshot 1`] = `
<div>
  <form>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </form>
  <table
    className="myClass"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
